tools/aa/commands/SyncCommand.java

```diff
@@ -105,9 +110,18 @@ public boolean run(String[] args) {
                 repo.removeBranch("temp_branch_for_sync");
               });
     } catch (IOException e) {
+      revertChanges(workspacePath, wasTempBranchForSyncExisted);
       e.printStackTrace();
     }
     return true;
   }
+
+  private void revertChanges(String workspacePath, boolean wasTempBranchForSyncExist) {
+    GitRepo repo = repoFactory.create(workspacePath);
+    ImmutableList<String> branches = repo.listBranches();
+    if (branches.contains(TEMP_BRANCH_FOR_SYNC) && !wasTempBranchForSyncExist) {
```

Comment:
Actually in this case, let's leave the temp branch, because we're not really undoing everything, and if there was an error, the temp branch may still have information that is otherwise lost.

We should, however, switch to the original branch.

Reply:
Done.